libs/math/tests/math/all_tests/tensor/tensor_operations.cpp

```diff
@@ -259,4 +259,171 @@ TYPED_TEST(TensorOperationsTest, slice_and_transpose_test)
+    {
+      for (std::uint64_t k{0}; k < 5; ++k)
+      {
+        t1.At(j, k, i) = TypeParam(count);
+        ++count;
+      }
+    }
+  }
+
+  auto t1s = t1.Slice(1, 2);
+  auto t2s = t1s.Slice(2, 1);
+
+  fetch::math::Tensor<TypeParam> t1t = t1s.Copy();
+  EXPECT_EQ(t1t.shape(), std::vector<std::uint64_t>({3, 5, 1}));
+
+  EXPECT_EQ(t1t.At(0, 0, 0), TypeParam(15));
+  EXPECT_EQ(t1t.At(1, 0, 0), TypeParam(20));
+  EXPECT_EQ(t1t.At(2, 0, 0), TypeParam(25));
+  EXPECT_EQ(t1t.At(0, 1, 0), TypeParam(16));
+  EXPECT_EQ(t1t.At(1, 1, 0), TypeParam(21));
+  EXPECT_EQ(t1t.At(2, 1, 0), TypeParam(26));
+  EXPECT_EQ(t1t.At(0, 2, 0), TypeParam(17));
+  EXPECT_EQ(t1t.At(1, 2, 0), TypeParam(22));
+  EXPECT_EQ(t1t.At(2, 2, 0), TypeParam(27));
+  EXPECT_EQ(t1t.At(0, 3, 0), TypeParam(18));
+  EXPECT_EQ(t1t.At(1, 3, 0), TypeParam(23));
+  EXPECT_EQ(t1t.At(2, 3, 0), TypeParam(28));
+  EXPECT_EQ(t1t.At(0, 4, 0), TypeParam(19));
+  EXPECT_EQ(t1t.At(1, 4, 0), TypeParam(24));
+  EXPECT_EQ(t1t.At(2, 4, 0), TypeParam(29));
+
+  fetch::math::Tensor<TypeParam> t2t = t2s.Copy();
+  EXPECT_EQ(t2t.shape(), std::vector<std::uint64_t>({3, 1, 1}));
+
+  EXPECT_EQ(t2t.At(0, 0, 0), TypeParam(17));
+  EXPECT_EQ(t2t.At(1, 0, 0), TypeParam(22));
+  EXPECT_EQ(t2t.At(2, 0, 0), TypeParam(27));
+}
+
+TYPED_TEST(TensorOperationsTest, multiple_const_slices_separated_test)
+{
+  fetch::math::Tensor<TypeParam> t1(std::vector<std::uint64_t>({3, 5, 2}));
+  std::uint64_t                  count = 0;
+  for (std::uint64_t i{0}; i < 2; ++i)
+  {
+    for (std::uint64_t j{0}; j < 3; ++j)
+    {
+      for (std::uint64_t k{0}; k < 5; ++k)
+      {
+        t1.At(j, k, i) = TypeParam(count);
+        ++count;
+      }
+    }
+  }
+
+  const fetch::math::Tensor<TypeParam> t1c = t1.Copy();
+
+  auto t1s = t1c.Slice(1, 2);
+  auto t2s = t1s.Slice(2, 1);
+
+  fetch::math::Tensor<TypeParam> t1t = t1s.Copy();
+  EXPECT_EQ(t1t.shape(), std::vector<std::uint64_t>({3, 5, 1}));
+
+  EXPECT_EQ(t1t.At(0, 0, 0), TypeParam(15));
+  EXPECT_EQ(t1t.At(1, 0, 0), TypeParam(20));
+  EXPECT_EQ(t1t.At(2, 0, 0), TypeParam(25));
+  EXPECT_EQ(t1t.At(0, 1, 0), TypeParam(16));
+  EXPECT_EQ(t1t.At(1, 1, 0), TypeParam(21));
+  EXPECT_EQ(t1t.At(2, 1, 0), TypeParam(26));
+  EXPECT_EQ(t1t.At(0, 2, 0), TypeParam(17));
+  EXPECT_EQ(t1t.At(1, 2, 0), TypeParam(22));
+  EXPECT_EQ(t1t.At(2, 2, 0), TypeParam(27));
+  EXPECT_EQ(t1t.At(0, 3, 0), TypeParam(18));
+  EXPECT_EQ(t1t.At(1, 3, 0), TypeParam(23));
+  EXPECT_EQ(t1t.At(2, 3, 0), TypeParam(28));
+  EXPECT_EQ(t1t.At(0, 4, 0), TypeParam(19));
+  EXPECT_EQ(t1t.At(1, 4, 0), TypeParam(24));
+  EXPECT_EQ(t1t.At(2, 4, 0), TypeParam(29));
+
+  fetch::math::Tensor<TypeParam> t2t = t2s.Copy();
+  EXPECT_EQ(t2t.shape(), std::vector<std::uint64_t>({3, 1, 1}));
+
+  EXPECT_EQ(t2t.At(0, 0, 0), TypeParam(17));
+  EXPECT_EQ(t2t.At(1, 0, 0), TypeParam(22));
+  EXPECT_EQ(t2t.At(2, 0, 0), TypeParam(27));
+}
+
+TYPED_TEST(TensorOperationsTest, multiple_slices_assign_test)
```

Comment:
nice series of tests.
Since slices are by reference, would it be a good idea to also have a test that checks something like the following:
t1s = t.Slice
t2s = t.Slice
t2s(0, 0) = 7
ExpectEq(t1s() == 7)

I guess you've partly done that in the multiple slices assign test?